File: server-ruby-sinatra/api/weather_api.rb
require 'json'


MyApp.add_route('GET', '/api/3.18/weather/search', {
  "resourcePath" => "/Weather",
  "summary" => "Search Weather",
  "nickname" => "search_weather",
  "responseClass" => "WeatherResponse",
  "endpoint" => "/weather/search",
  "notes" => "Search the weather forcast for the next 5 days",
  "parameters" => [
    {
      "name" => "region_id",
      "description" => "Region Id",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "latitude",
      "description" => "Latitude",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "Longitude",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "timezone_offset",
      "description" => "Timezone Offset",
      "dataType" => "Integer",
      "allowableValues" => "",
      "defaultValue" => "-6",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

File: server-scala-play-server/app/api/WeatherApi.scala
package api

import play.api.libs.json._
import model.WeatherResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-11T19:31:43.310890579Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
trait WeatherApi {
  /**
    * Search Weather
    * Search the weather forcast for the next 5 days
    * @param regionId Region Id
    * @param latitude Latitude
    * @param longitude Longitude
    * @param timezoneOffset Timezone Offset
    */
  def searchWeather(regionId: Option[Long], latitude: Option[Double], longitude: Option[Double], timezoneOffset: Option[Long]): WeatherResponse
}
